### ISSUE TITLE
ghost update fails to restart

### ISSUE THREAD

**Issue by jloh** [TryGhost/Ghost-CLI#430]
Second time I've run into this one (originally thought it'd be fixed in the latest version of Ghost CLI) however it seems it fails to restart my ghost install after upgrading it. Restarting it again with `systemd` works fine however.

```
root@us # ghost upgrade
✔ Checking for latest Ghost version
✔ Downloading and updating Ghost to v1.5.0
ℹ Stopping Ghost [skipped]
✔ Linking latest Ghost and recording versions
Running sudo command: -E -u ghost /usr/lib/node_modules/ghost-cli/node_modules/.bin/knex-migrator-migrate --init --mgpath /opt/ghost/bytebybyte/current
[2017-08-04 16:06:02] INFO Finished database migration!
✔ Running database migrations
✖ Validating config
✖ Restarting Ghost
A ConfigError occurred.

Error detected in the production configuration.

Message: Port '2369' is in use.
Configuration Key(s): server.port
Current Value(s): 2369

Help:  Run `ghost config <key> <new value>` for each key to fix the issue.


Debug Information:
    Node Version: v6.11.0
    Ghost-CLI Version: 1.0.3
    Environment: production
    Command: 'ghost upgrade'

Please refer to https://docs.ghost.org/docs/troubleshooting#section-cli-errors for troubleshooting.
```

I'm using Ubuntu 16.04. It looks like the cli can't see the current running ghost instance?

I have two installs currently on this machine, a ghost-cli one and a ghost 0.11.x install.

The above instance was fully installed using the ghost-cli, so the `systemd` script should be where it is expected (the symlink is still intact). Whats interesting is that `ghost status` doesn't seem to think its running:

```
root@us :/opt/ghost/bytebybyte# ghost status
┌─────────────────────┬───────────────────────┬─────────┬─────────┬─────┬──────┬─────────────────┐
│ Name                │ Location              │ Version │ Status  │ URL │ Port │ Process Manager │
├─────────────────────┼───────────────────────┼─────────┼─────────┼─────┼──────┼─────────────────┤
│ ghost-bytebybyte-io │ /opt/ghost/bytebybyte │ 1.5.0   │ stopped │ n/a │ n/a  │ n/a             │
└─────────────────────┴───────────────────────┴─────────┴─────────┴─────┴──────┴─────────────────┘
root@us :/opt/ghost/bytebybyte#
```

But systemd shows it as running (and it works in my browser).

`ghost-cli` file for reference:

```json
{
  "cli-version": "1.0.0",
  "active-version": "1.5.0",
  "name": "ghost-bytebybyte-io",
  "extension": {
    "systemd": true
  },
  "previous-version": "1.4.0"
}
```

Sorry if this has already been reported!

**Comment by acburdine** [comment]
Just out of curiosity -> before you ran `ghost update` do you know if `ghost ls` showed whether or not Ghost was running?

**Comment by acburdine** [comment]
I could see this happening (related to #431) if you rebooted your machine and then ran `ghost update` sometime after that. The above log is showing that update didn't try to stop ghost before re-linking things, which makes me think that's what's happening.

**Comment by jloh** [comment]
@acburdine yeah I'm pretty sure I did kernel updates (and thus reboots) between the installation and reproducing this issue for the second time.

I believe I've also generally restarted/started Ghost via the systemd command as opposed to using the ghost cli utility (primarily just using it for updates and installations.

**Comment by linsyorozuya** [comment]
I'm also having exactly the same issue.....
![b51d59fd-898d-4083-b215-a4f11d58de24](https://user-images.githubusercontent.com/4980617/29449866-01e1e504-842f-11e7-87e4-0eebe94cf192.png)

**Comment by alznyc** [comment]
Has anyone figured out a solution to this. I'm having the same issue.

Ghost update will quit at Validating config

A ConfigError occurred.

Error detected in the production configuration.

Message: Port '2369' is in use.
Configuration Key(s): server.port
Current Value(s): 2369

Help: Run `ghost config <key> <new value>` for each key to fix the issue.

Debug Information:
    Node Version: v6.9.1
    Ghost-CLI Version: 1.1.1
    Environment: production
    Command: 'ghost update'


ghost@krypton:/var/www/XXXX$ ghost version
Ghost-CLI version: 1.1.1
Ghost Version (at /var/www/XXXX): 1.7.0

**Comment by alznyc** [comment]
If anyone still needs help with this...here is the solution that worked for me:
run:
sudo systemctl stop {ghost service) (use ls -la system/files to find the name of your ghost instance)

cd /var/www/ghost
ghost restart

**Comment by gmschroeder** [comment]
Same problem as above:
| Name         │ Location            │ Version│ Status    │ URL │ Port │ Process Manager │
│ greg-party │ /var/www/ghost │ 1.16.0  │ stopped │ n/a    │ n/a  │ n/a

**Comment by acburdine** [comment]
Closed via #607
